src/components/Image/index.native.tsx

```diff
@@ -1,35 +1,31 @@
-import lodashGet from 'lodash/get';
 import React from 'react';
 import RNFastImage from 'react-native-fast-image';
 import {withOnyx} from 'react-native-onyx';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import {defaultProps, imagePropTypes} from './imagePropTypes';
 import RESIZE_MODES from './resizeModes';
+import {DimensionsCacheValue, FastImageSource, ImageOnyxProps, ImagePropsWithOnyx} from './types';
 
-const dimensionsCache = new Map();
+const dimensionsCache = new Map<string, DimensionsCacheValue>();
 
-function resolveDimensions(key) {
+function resolveDimensions(key: string): DimensionsCacheValue | undefined {
     return dimensionsCache.get(key);
 }
 
-function Image(props) {
-    // eslint-disable-next-line react/destructuring-assignment
-    const {source, isAuthTokenRequired, session, ...rest} = props;
-
-    let imageSource = source;
-    if (source && source.uri && typeof source.uri === 'number') {
+function Image({source, isAuthTokenRequired, session, ...rest}: ImagePropsWithOnyx) {
+    let imageSource: FastImageSource = source;
+    if (typeof source !== 'number' && typeof source.uri === 'number') {
```

Comment:
I'm confused about this change, it was `if (source && source.uri && typeof source.uri === 'number') {`, but you changed condition a little bit. Why?

Reply:
Sure, let me explain! `source` can either be a number, or an object that potentially has an `uri` key. It's mostly a matter of correct order inside this condition. Source can be a require path (a number) so you can't start by checking for keys, first you have to make sure it's not a number. So the new condition goes as follows:
- Make sure `source` is not a number (equivalent of `source` from the original if statement)
- Check if `uri` is a number (equivalent of `source.uri && typeof source.uri === 'number'` from the original if statement)

Let me know if that answered your question!

Reply:
makes sense, thanks!